mne/decoding/tests/test_receptive_field.py

```diff
@@ -144,7 +145,93 @@ def test_receptive_field():
+    """Test that the fast solving works like Ridge."""
+    from sklearn.linear_model import Ridge
+    rng = np.random.RandomState(0)
+    y = np.zeros(500)
+    x = rng.randn(500)
+    for delay in range(-2, 3):
+        y.fill(0.)
+        slims = [(-4, 2)]
+        if delay == 0:
+            y = x.copy()
+        elif delay < 0:
+            y[-delay:] = x[:delay]
+            slims += [(-4, -1)]
+        else:
+            y[:-delay] = x[delay:]
+            slims += [(1, 2)]
+        for slim in slims:
+            tdr = TimeDelayingRidge(slim[0], slim[1], 1., 0.1, 'quadratic',
+                                    fit_intercept=False)
+            for estimator in (Ridge(alpha=0.), 0., 0.1, tdr):
+                model = ReceptiveField(slim[0], slim[1], 1.,
+                                       estimator=estimator)
+                model.fit(x[:, np.newaxis], y)
+                assert_array_equal(model.delays_,
+                                   np.arange(slim[0], slim[1] + 1))
+                expected = (model.delays_ == delay).astype(float)
+                assert_allclose(model.coef_[0], expected, atol=1e-2)
+                p = model.predict(x[:, np.newaxis])[:, 0]
+                assert_allclose(y, p, atol=5e-2)
+    # multidimensional
+    x = rng.randn(1000, 3)
+    y = np.zeros((1000, 2))
+    slim = [-5, 0]
+    # This is a weird assignment, but it's just a way to distribute some
+    # unique values at various delays, and "expected" explains how they
+    # should appear in the resulting RF
+    for ii in range(1, 5):
+        y[ii:, ii % 2] += (-1) ** ii * ii * x[:-ii, ii % 3]
+    expected = [
+        [[0, 0, 0, 0, 0, 0],
+         [0, 4, 0, 0, 0, 0],
+         [0, 0, 0, 2, 0, 0]],
+        [[0, 0, -3, 0, 0, 0],
+         [0, 0, 0, 0, -1, 0],
+         [0, 0, 0, 0, 0, 0]],
+    ]
+    tdr = TimeDelayingRidge(slim[0], slim[1], 1., 0.1, 'quadratic')
+    for estimator in (Ridge(alpha=0.), 0., 0.01, tdr):
+        model = ReceptiveField(slim[0], slim[1], 1.,
+                               estimator=estimator)
+        model.fit(x, y)
+        assert_array_equal(model.delays_,
+                           np.arange(slim[0], slim[1] + 1))
+        assert_allclose(model.coef_, expected, atol=1e-1)
+    tdr = TimeDelayingRidge(slim[0], slim[1], 1., 0.01, reg_type='foo')
+    model = ReceptiveField(slim[0], slim[1], 1., estimator=tdr)
+    assert_raises(ValueError, model.fit, x, y)
+    tdr = TimeDelayingRidge(slim[0], slim[1], 1., 0.01, reg_type=['quadratic'])
+    model = ReceptiveField(slim[0], slim[1], 1., estimator=tdr)
+    assert_raises(ValueError, model.fit, x, y)
+    # Now check the intercept_
+    tdr = TimeDelayingRidge(slim[0], slim[1], 1., 0.)
+    tdr_no = TimeDelayingRidge(slim[0], slim[1], 1., 0., fit_intercept=False)
+    for estimator in (Ridge(alpha=0.), tdr,
+                      Ridge(alpha=0., fit_intercept=False), tdr_no):
+        x -= np.mean(x, axis=0)
+        y -= np.mean(y, axis=0)
```

Comment:
> Results should match for Ridge and TimeDelayingRidge with and without intercept whatever if the data have been centered or not.

I don't think the results were the same even for `Ridge` with `fit_intercept=True` vs `fit_intercept=False`, but I'll double-check.

> I am asking as I see slightly different values for the intercept_ when I switch base estimators.

I think this might be due to the differences in behavior with cropping to "valid" samples, but I'll try to confirm that, too.